src/java/io/bazel/rulesscala/scalac/ScalacProcessor.java

```diff
@@ -58,6 +58,9 @@ public void processRequest(List<String> args) throws Exception {
       List<File> javaJarFiles = filterFilesByExtension(jarFiles, ".java");
 
       String[] scalaSources = GenericWorker.appendToString(ops.files, scalaJarFiles);
+      // include java files from .srcjar's to support mixed compilation
+      scalaSources = GenericWorker.appendToString(scalaSources, javaJarFiles);
```

Comment:
How about extracting the new line and the existing one to "appendSrcJarSources"? I think inside that context we can live without the comment. Wdyt?